framework/src/executioners/Eigenvalue.C

```diff
@@ -153,6 +154,16 @@ Eigenvalue::Eigenvalue(const InputParameters & parameters)
   mooseDeprecated(
       "Please use SLEPc-3.13.0 or higher. Old versions of SLEPc likely produce bad convergence");
 #endif
+
+  // To avoid petsc unused option warnings, ensure we do not set irrelevant options.
+  Moose::PetscSupport::dontAddLinearConvergedReason(_fe_problem);
+  Moose::PetscSupport::dontAddNonlinearConvergedReason(_fe_problem);
+  Moose::PetscSupport::dontAddPetscFlag("-mat_mffd_type", _fe_problem.getPetscOptions());
+  Moose::PetscSupport::dontAddPetscFlag("-st_ksp_atol", _fe_problem.getPetscOptions());
+  Moose::PetscSupport::dontAddPetscFlag("-st_ksp_max_it", _fe_problem.getPetscOptions());
+  Moose::PetscSupport::dontAddPetscFlag("-st_ksp_rtol", _fe_problem.getPetscOptions());
+  if (!_eigen_problem.solverParams()._eigen_matrix_free)
+    Moose::PetscSupport::dontAddPetscFlag("-snes_mf_operator", _fe_problem.getPetscOptions());
```

Comment:
Now if a user adds these from their input file or the command line, those options will be respected right?

Reply:
Correct. That check is done in `PetscSetOptionsHelper`: https://github.com/idaholab/moose/blob/770678ac586b70c2e51bf1ec2a378080c2ce19fb/framework/src/utils/PetscSupport.C#L259